### PR TITLE
system_tests for add/replace blobber

### PR DESCRIPTION
**Pull request by boddumanohar**
associated PRs: 

zboxcli: https://github.com/0chain/zboxcli/pull/401
gosdk: https://github.com/0chain/gosdk/pull/908

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | add-replace-blobber  |
| gosdk  | update-alloc  |
| zboxcli  | add-replace-blobber  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | add-replace-blobber  |
| gosdk  | staging  |
| zboxcli  | add-replace-blobber  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | add-replace-blobber  |
| gosdk  | staging  |
| zboxcli  | add-replace-blobber  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Review by stewartie4 — COMMENTED** [review]
@boddumanohar what's the status of this PR?

**Comment by peterlimg** [comment]
@boddumanohar any progress on this? do you have any question about the system test failures

File: tests/cli_tests/sdk.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	"os"
+
+	"github.com/0chain/gosdk/core/common"
+	"github.com/0chain/gosdk/zboxcore/fileref"
+	"github.com/0chain/gosdk/zboxcore/sdk"
+	"github.com/0chain/system_test/internal/api/util/config"
+	"github.com/0chain/system_test/internal/api/util/crypto"
+)
+
+// getBlobberNotPartOfAllocation returns a blobber not part of current allocation
```
Comment (Kishan-Dhakan) [inline tests/cli_tests/sdk.go:17]: This comment should be on line 41

File: tests/cli_tests/zboxcli_update_allocation_test.go [hunk 2]
```diff
@@ -826,6 +828,103 @@ func TestUpdateAllocation(testSetup *testing.T) {
 		// assert that no more blobber was added
 		require.Equal(t, len(alloc.Blobbers), len(updatedAlloc.Blobbers))
 	})
+	t.Run("Update allocation with add blobber should succeed", func(t *test.SystemTest) {
```
Comment (Kishan-Dhakan) [inline tests/cli_tests/zboxcli_update_allocation_test.go:831]: Can have a newline b/w cases for readability

**Review by Kishan-Dhakan — APPROVED** [review]
LGTM, minor style changes requested

**Comment by Kishan-Dhakan** [comment]
Happy to merge pending a passing run

**Review by stewartie4 — COMMENTED** [review]
@boddumanohar can we merge this PR?

**Comment by boddumanohar** [comment]
```
    utils.go:67: Command failed on attempt [1/3] due to error [exit status 1]. Output: [Allocation updated with txId : 5001226803cbf7db0b76bf3b592bf0f269348ac238878caf9be7c86936dc98d7 -<NEWLINE>- 0 / 1024    0.00% 0 / 1024    0.00% 0s -<NEWLINE>- Error in file operation: Download failed for block 1. -<NEWLINE>- invalid_data: some blobber responded with wrong data]
```

At the moment, facing this issue. Currently looking into it.

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | add-replace-blobber  |
| gosdk  | update-alloc  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Review by peterlimg — APPROVED** [review]

**Pull request merged**